lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -630,6 +630,11 @@ class GenerateKernelExecution
       return;
     }
 
+    // After kernel launch, we clean up any dangling arrays that are allocated
+    // during kernel execution.
+    builder.create<func::CallOp>(loc, std::nullopt,
+                                 cudaq::runtime::cleanupArrays, ValueRange{});
```

Comment:
We'll need this in Python as well, since the leaking allocation is at the C layer underneath the interpreter.

Reply:
I don't see why it needs to be in both functions. Arguably putting it in the host side code (which is being done here) isn't really correct, since the device isn't the host.

Reply:
I see. Fixed in https://github.com/NVIDIA/cuda-quantum/pull/3478/commits/75e60570f1854a9483a2d2d87f1956d75071e243 (removed this function, just kept the one in `thunk`)